src/Internal/Payment/State/SystemErrorState.php

```diff
@@ -0,0 +1,18 @@
+<?php
+/**
+ * Class SystemErrorState
+ *
+ * @package WooCommerce\Payments
+ */
+
+namespace WCPay\Internal\Payment\State;
+
+/**
+ * Erroneous state, caused by a bug or an edge case.
+ *
+ * Consider this state an exception: Ideally it would never be entered,
+ * but in case any sort of software issue arrises, it will be used.
+ */
+class SystemErrorState extends PaymentState {
```

Comment:
I like custom exceptions, but I think that we could add custom error codes and messages, so we don't have to manually add them when the exception is handled.

Reply:
This will also be a stream of work.

Keep in mind, that I have questions here myself:

1. How do we transfer the exception to all necessary layers, including the context and the caller?
2. Which exceptions, and how, would be directed to one state vs the other?
3. Do we store the exception in the database, or simply log the type of error it is (payment vs system)?

We'll get there :)